### DISCUSSION TITLE
Building Fast-CDR failed on windows using MinGW [13749]

### DISCUSSION THREAD

**Discussion by sky-fire** [eProsima/Fast-CDR#166] [Unofficial support]
I followed the instructions [here](https://fast-dds.docs.eprosima.com/en/latest/installation/sources/sources_windows.html), but when I tap this command: cmake --build . --target install, it failed,and the fail reason is : /usr/bin/windres:no resources

**Comment by MiguelCompany** [comment]
@sky-fire 

It seems you are not building natively on windows, as `/usr/bin/windres` seems a linux-like path.

The instructions are for natively building on a system with Visual Studio installed.

So can you fully describe your build environment?

**Comment by LanceAZA** [comment]
I have the same problem by using MinGW in windows

**Comment by JLBuenoLopez** [comment]
MinGW is not an officially supported platform. I am moving the ticket to the corresponding discussion forum.

**Comment by felixfeixu** [comment]
This is actually a valid issue.

The error happens when building FastCdr as a static library, in that case `fastcdr_EXPORTS` is not defined, then

https://github.com/eProsima/Fast-CDR/blob/f889eee5880f521931214a1f8aa11efcd003b421/src/cpp/FastCdr.rc#L86-L121

will be empty, that's why windres says: 
>no resources

**Comment by stefanct** [comment]
I can confirm felix' observations. I am also building fast-cdr by use of mingw as part of a bigger cmake project (plotjuggler) which defaults to building fast-cdr statically AFAICT. If I comment out the ifdef, the build of fast-cdr succeeds and an application linked to it can also be built successfully. So I agree with felix that this is an actual issue that would be nice to solve as it probably is also not correct for non-mingw builds (even though it might simply be ignored by the microsoft tools). The check was changed in `e251dea` by @richiware. I have no idea about resource files but if the whole file is not needed in static builds then the correct fix would be to not add it to `${PROJECT_NAME}_source_files` in the cmake file if not building the shared lib.

I have submitted a PR that implements this fix: https://github.com/eProsima/Fast-CDR/pull/259
